File: python/leetcode/question49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        strs_dict:Dict[str,List[str]] = {}
        for str in strs:
            str_sort = "".join(sorted(str))
            if str_sort in strs_dict:
                strs_dict[str_sort].append(str)
            else :
                strs_dict[str_sort] = [str]
        return list(strs_dict.values())

File: typescript/leetcode/question49.ts
function groupAnagrams(strs: string[]): string[][] {
    const strs_dict = new Map<string, string[]>();
    for (let index = 0; index < strs.length; index++) {
        const sortedStr = strs[index].split("").sort().join("");
        if (strs_dict.has(sortedStr)){
            strs_dict.get(sortedStr)?.push(strs[index]);
        }else{
            strs_dict.set(sortedStr,[strs[index]]);
        }
    }
    return Array.from(strs_dict.values());
};